server/webapp/WEB-INF/rails.new/app/controllers/api_v1/users_controller.rb

```diff
@@ -43,12 +37,41 @@ def update
       end
     end
 
+    def create
+      @user_to_operate = user_service.findUserByName(params[:login_name])
+
+      unless @user_to_operate.instance_of?(com.thoughtworks.go.domain.NullUser)
```

Comment:
I'm aware of this problem with possible race condition. However `addUserIfDoesNotExist` is not what I'm looking to do – I want to raise an error if a user already exists, or create if none exists, unless you mean to use `enableUserMutex` to acquire a lock.